File: novelvids/api/v1/endpoints/storyboard.py
"""分镜 API 接口。

使用任务模式进行分镜生成，支持轮询查询进度。
"""

import logging
from uuid import UUID

from fastapi import APIRouter, BackgroundTasks, Depends, HTTPException, status
from pydantic import BaseModel, Field

from novelvids.api.dependencies import get_current_user
from novelvids.application.services.storyboard_task_service import storyboard_task_service
from novelvids.domain.models.storyboard import Shot
from novelvids.domain.services.storyboard import create_storyboard_service
from novelvids.domain.services.storyboard.prompts import SUPPORTED_VIDEO_PLATFORMS
from novelvids.core.config.settings import get_settings
from novelvids.infrastructure.database.models import (
    ChapterModel,
    StoryboardTaskModel,
    TaskStatus,
    UserModel,
)

logger = logging.getLogger(__name__)
router = APIRouter(prefix="/storyboard", tags=["分镜"])


# ============== DTO ==============


class StoryboardGenerateDTO(BaseModel):
    """分镜生成请求 DTO。"""

    max_shot_duration: float = Field(default=8.0, ge=4.0, le=15.0, description="单个镜头最大时长")
    target_platform: str = Field(default="vidu", description="目标平台：vidu/doubao")
    style_preset: str = Field(default="cinematic", description="风格预设")
    aspect_ratio: str = Field(default="16:9", description="画面比例")
    include_audio: bool = Field(default=True, description="是否包含音频指令")


class TaskResponseDTO(BaseModel):
    """任务响应 DTO。"""

    id: UUID
    chapter_id: UUID
    status: str
    progress: int
    message: str | None
    error: str | None = None
    result: dict | None = None
    created_at: str
    started_at: str | None = None
    completed_at: str | None = None

    class Config:
        from_attributes = True


class ShotUpdateDTO(BaseModel):
    """分镜更新 DTO。"""

    name: str | None = None
    description_cn: str | None = None
    camera: dict | None = None
    subject: dict | None = None
    environment: dict | None = None
    style: dict | None = None
    audio: dict | None = None
    technical: dict | None = None
    negative: dict | None = None
    transition_in: str | None = None
    transition_out: str | None = None


class StoryboardResponseDTO(BaseModel):
    """分镜响应 DTO。"""

    chapter_id: UUID
    chapter_number: int
    chapter_title: str
    shots: list[dict]
    total_duration: float
    shot_count: int

    class Config:
        from_attributes = True


class ShotPromptDTO(BaseModel):
    """分镜提示词响应 DTO。"""

    sequence: int
    prompt: str
    negative_prompt: str
    platform: str


class StoryboardPromptsDTO(BaseModel):
    """完整分镜提示词响应 DTO。"""

    chapter_id: UUID
    platform: str
    prompts: list[ShotPromptDTO]


# ============== 辅助函数 ==============


def _task_to_dto(task: StoryboardTaskModel) -> TaskResponseDTO:
    """将任务模型转换为 DTO。"""
    return TaskResponseDTO(
        id=task.id,
        chapter_id=task.chapter_id,
        status=task.status.value,
        progress=task.progress,
        message=task.message,
        error=task.error,
        result=task.result,
        created_at=task.created_at.isoformat(),
        started_at=task.started_at.isoformat() if task.started_at else None,
        completed_at=task.completed_at.isoformat() if task.completed_at else None,
    )


async def _verify_chapter_access(chapter_id: UUID, current_user: UserModel) -> ChapterModel:
    """验证章节访问权限。"""
    chapter = await ChapterModel.filter(id=chapter_id).prefetch_related("novel").first()
    if not chapter:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Chapter not found")

    novel = await chapter.novel
    if novel.user_id != current_user.id:
        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Access denied")

    return chapter


# ============== 任务 API 端点 ==============


@router.post("/chapters/{chapter_id}/generate", response_model=TaskResponseDTO)
async def start_generate_storyboard(
    chapter_id: UUID,
    request: StoryboardGenerateDTO,
    background_tasks: BackgroundTasks,
    current_user: UserModel = Depends(get_current_user),
):
    """启动分镜生成任务。

    返回任务ID，前端通过轮询 GET /tasks/{task_id} 查询进度。
    
    仅支持 vidu 和 doubao 平台。
    """
    # 验证平台
    if request.target_platform not in SUPPORTED_VIDEO_PLATFORMS:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=f"不支持的平台: {request.target_platform}，仅支持: {SUPPORTED_VIDEO_PLATFORMS}",
        )

    await _verify_chapter_access(chapter_id, current_user)

    # 创建任务
    task = await storyboard_task_service.create_task(
        chapter_id=chapter_id,
        target_platform=request.target_platform,
        max_shot_duration=request.max_shot_duration,
        style_preset=request.style_preset,
        aspect_ratio=request.aspect_ratio,
        include_audio=request.include_audio,
    )

    # 如果是新任务，启动后台执行
    if task.status == TaskStatus.PENDING:
        background_tasks.add_task(storyboard_task_service.execute_task, task.id)

    return _task_to_dto(task)


@router.get("/tasks/{task_id}", response_model=TaskResponseDTO)
async def get_task_status(
    task_id: UUID,
    current_user: UserModel = Depends(get_current_user),
):
    """获取分镜生成任务状态。"""
    task = await storyboard_task_service.get_task(task_id)
    if not task:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Task not found")

    # 验证权限
    await _verify_chapter_access(task.chapter_id, current_user)

    return _task_to_dto(task)


@router.get("/chapters/{chapter_id}/task", response_model=TaskResponseDTO | None)
async def get_chapter_task(
    chapter_id: UUID,
    current_user: UserModel = Depends(get_current_user),
):
    """获取章节的最新分镜生成任务。"""
    await _verify_chapter_access(chapter_id, current_user)

    task = await storyboard_task_service.get_chapter_task(chapter_id)
    if not task:
        return None

    return _task_to_dto(task)


# ============== 分镜数据 API 端点 ==============


@router.get("/chapters/{chapter_id}", response_model=StoryboardResponseDTO)
async def get_storyboard(
    chapter_id: UUID,
    current_user: UserModel = Depends(get_current_user),
):
    """获取章节的分镜脚本。"""
    chapter = await _verify_chapter_access(chapter_id, current_user)

    storyboard_data = chapter.metadata.get("storyboard")

    # 没有分镜时返回空数组
    if not storyboard_data:
        return StoryboardResponseDTO(
            chapter_id=chapter_id,
            chapter_number=chapter.number,
            chapter_title=chapter.title,
            shots=[],
            total_duration=0,
            shot_count=0,
        )

    return StoryboardResponseDTO(
        chapter_id=chapter_id,
        chapter_number=chapter.number,
        chapter_title=chapter.title,
        shots=storyboard_data.get("shots", []),
        total_duration=storyboard_data.get("total_duration", 0),
        shot_count=len(storyboard_data.get("shots", [])),
    )


@router.put("/chapters/{chapter_id}/shots/{sequence}", response_model=dict)
async def update_shot(
    chapter_id: UUID,
    sequence: int,
    update: ShotUpdateDTO,
    current_user: UserModel = Depends(get_current_user),
):
    """更新单个分镜。"""
    chapter = await _verify_chapter_access(chapter_id, current_user)

    storyboard_data = chapter.metadata.get("storyboard")
    if not storyboard_data:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Storyboard not found")

    # 查找并更新镜头
    shots = storyboard_data.get("shots", [])
    shot_index = None
    for i, shot in enumerate(shots):
        if shot.get("sequence") == sequence:
            shot_index = i
            break

    if shot_index is None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"Shot {sequence} not found")

    # 更新字段
    update_dict = update.model_dump(exclude_unset=True)
    for key, value in update_dict.items():
        if value is not None:
            if isinstance(value, dict):
                shots[shot_index][key] = {**shots[shot_index].get(key, {}), **value}
            else:
                shots[shot_index][key] = value

    # 保存
    chapter.metadata["storyboard"]["shots"] = shots
    await chapter.save()

    return shots[shot_index]


@router.delete("/chapters/{chapter_id}/shots/{sequence}")
async def delete_shot(
    chapter_id: UUID,
    sequence: int,
    current_user: UserModel = Depends(get_current_user),
):
    """删除单个分镜。"""
    chapter = await _verify_chapter_access(chapter_id, current_user)

    storyboard_data = chapter.metadata.get("storyboard")
    if not storyboard_data:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Storyboard not found")

    shots = storyboard_data.get("shots", [])
    new_shots = [s for s in shots if s.get("sequence") != sequence]

    if len(new_shots) == len(shots):
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"Shot {sequence} not found")

    # 重新编号
    for i, shot in enumerate(new_shots, start=1):
        shot["sequence"] = i

    total_duration = sum(s.get("technical", {}).get("duration", 0) for s in new_shots)

    chapter.metadata["storyboard"]["shots"] = new_shots
    chapter.metadata["storyboard"]["total_duration"] = total_duration
    chapter.scene_count = len(new_shots)
    await chapter.save()

    return {"message": "Shot deleted", "shot_count": len(new_shots)}


@router.post("/chapters/{chapter_id}/shots", response_model=dict)
async def add_shot(
    chapter_id: UUID,
    shot: dict,
    after_sequence: int | None = None,
    current_user: UserModel = Depends(get_current_user),
):
    """添加新分镜。"""
    chapter = await _verify_chapter_access(chapter_id, current_user)

    storyboard_data = chapter.metadata.get("storyboard")
    if not storyboard_data:
        storyboard_data = {"shots": [], "total_duration": 0}
        chapter.metadata["storyboard"] = storyboard_data

    shots = storyboard_data.get("shots", [])

    insert_index = len(shots) if after_sequence is None else after_sequence
    shot["sequence"] = insert_index + 1

    shots.insert(insert_index, shot)

    for i, s in enumerate(shots, start=1):
        s["sequence"] = i

    total_duration = sum(s.get("technical", {}).get("duration", 0) for s in shots)

    chapter.metadata["storyboard"]["shots"] = shots
    chapter.metadata["storyboard"]["total_duration"] = total_duration
    chapter.scene_count = len(shots)
    await chapter.save()

    return shot


@router.get("/chapters/{chapter_id}/prompts", response_model=StoryboardPromptsDTO)
async def get_storyboard_prompts(
    chapter_id: UUID,
    platform: str = "vidu",
    current_user: UserModel = Depends(get_current_user),
):
    """获取分镜的视频生成提示词。

    仅支持 vidu 和 doubao 平台。
    如果 shot 中已有预生成的 platform_prompt，则直接使用；否则实时构建。
    """
    # 验证平台
    if platform not in SUPPORTED_VIDEO_PLATFORMS:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=f"不支持的平台: {platform}，仅支持: {SUPPORTED_VIDEO_PLATFORMS}",
        )

    chapter = await _verify_chapter_access(chapter_id, current_user)

    storyboard_data = chapter.metadata.get("storyboard")
    if not storyboard_data:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Storyboard not found")

    prompts = []
    for shot_data in storyboard_data.get("shots", []):
        try:
            shot = Shot.model_validate(shot_data)

            # 优先使用预生成的 platform_prompt（已包含 {ref:id} 占位符）
            if shot.platform_prompt:
                prompt = shot.platform_prompt
            else:
                # 回退：实时构建（但不包含资产引用）
                settings_obj = get_settings()
                service = create_storyboard_service(
                    api_key=settings_obj.llm.api_key,
                    base_url=settings_obj.llm.base_url,
                    model_name=settings_obj.llm.model_name,
                )
                prompt = service.build_platform_prompt(shot=shot, platform=platform)

            negative_prompt = shot.build_negative_prompt()

            prompts.append(
                ShotPromptDTO(
                    sequence=shot.sequence,
                    prompt=prompt,
                    negative_prompt=negative_prompt,
                    platform=platform,
                )
            )
        except Exception as e:
            logger.warning(f"Failed to build prompt for shot {shot_data.get('sequence')}: {e}")

    return StoryboardPromptsDTO(
        chapter_id=chapter_id,
        platform=platform,
        prompts=prompts,
    )

File: frontend/src/api/storyboard.ts
/**
 * 分镜 API 客户端
 *
 * 使用任务模式进行分镜生成，支持轮询查询进度
 */

import api from './client'

// ============== 任务类型定义 ==============

/** 任务状态 */
export type TaskStatus = 'pending' | 'queued' | 'running' | 'completed' | 'failed' | 'cancelled'

/** 任务响应 */
export interface StoryboardTask {
  id: string
  chapter_id: string
  status: TaskStatus
  progress: number
  message: string | null
  error: string | null
  result: {
    shot_count?: number
    total_duration?: number
    warnings?: string[]
    retry_task_id?: string
  } | null
  created_at: string
  started_at: string | null
  completed_at: string | null
}

// ============== 分镜类型定义 ==============

/** 分镜生成请求 */
export interface StoryboardGenerateRequest {
  max_shot_duration?: number
  target_platform?: 'vidu' | 'doubao'
  style_preset?: string
  aspect_ratio?: string
  include_audio?: boolean
}

/** 相机设置 */
export interface CameraSettings {
  shot_size: string
  camera_angle: string
  camera_movement: string
  movement_speed: string
  lens_type: string
  depth_of_field: string
  focus_target: string | null
}

/** 主体与动作 */
export interface SubjectAction {
  subject_type: 'person' | 'scene' | 'item' | 'multiple'
  subject_description: string
  asset_refs: string[]
  action: string
  action_intensity: string
  emotion: string | null
  body_language: string | null
}

/** 环境设置 */
export interface EnvironmentSettings {
  location: string
  scene_asset_ref: string | null
  time_of_day: string
  weather: string | null
  lighting: string
  lighting_details: string | null
  atmosphere_elements: string[]
}

/** 风格设置 */
export interface StyleSettings {
  video_style: string
  mood: string
  color_grading: string | null
  film_grain: boolean
  contrast: string
  saturation: string
}

/** 音频设置 */
export interface AudioSettings {
  dialogue: string | null
  dialogue_speaker: string | null
  dialogue_tone: string | null
  sound_effects: string[]
  ambient_sounds: string[]
  background_music: string | null
  music_volume: string
}

/** 技术参数 */
export interface TechnicalSettings {
  duration: number
  aspect_ratio: string
  motion_speed: string
  resolution: string
  fps: number
}

/** 负面提示词 */
export interface NegativePrompt {
  avoid_elements: string[]
  avoid_artifacts: boolean
  avoid_text: boolean
}

/** 单个分镜 */
export interface Shot {
  sequence: number
  name: string | null
  description_cn: string
  source_text: string | null
  source_line_start: number | null
  source_line_end: number | null
  camera: CameraSettings
  subject: SubjectAction
  environment: EnvironmentSettings
  style: StyleSettings
  audio: AudioSettings
  technical: TechnicalSettings
  negative: NegativePrompt
  reference_images: string[]
  start_frame: string | null
  end_frame: string | null
  transition_in: string | null
  transition_out: string | null
  // 平台优化提示词
  platform_prompt: string | null
  // 视频生成任务状态
  video_task_id: string | null
  video_task_platform: string | null
  video_task_status: 'pending' | 'processing' | 'running' | 'queued' | 'success' | 'completed' | 'succeeded' | 'failed' | null
  video_task_progress: number
  video_url: string | null
  video_error: string | null
}

/** 分镜脚本响应 */
export interface StoryboardResponse {
  chapter_id: string
  chapter_number: number
  chapter_title: string
  shots: Shot[]
  total_duration: number
  shot_count: number
}

/** 分镜更新请求 */
export interface ShotUpdateRequest {
  name?: string
  description_cn?: string
  camera?: Partial<CameraSettings>
  subject?: Partial<SubjectAction>
  environment?: Partial<EnvironmentSettings>
  style?: Partial<StyleSettings>
  audio?: Partial<AudioSettings>
  technical?: Partial<TechnicalSettings>
  negative?: Partial<NegativePrompt>
  transition_in?: string
  transition_out?: string
}

/** 分镜提示词 */
export interface ShotPrompt {
  sequence: number
  prompt: string
  negative_prompt: string
  platform: string
}

/** 分镜提示词响应 */
export interface StoryboardPromptsResponse {
  chapter_id: string
  platform: string
  prompts: ShotPrompt[]
}

// ============== 任务 API 函数 ==============

/**
 * 启动分镜生成任务
 * 返回任务对象，前端通过轮询 getTaskStatus 查询进度
 */
export async function startGenerateStoryboard(
  chapterId: string,
  request: StoryboardGenerateRequest = {}
): Promise<StoryboardTask> {
  const response = await api.post(`/storyboard/chapters/${chapterId}/generate`, request)
  return response.data
}

/**
 * 获取任务状态
 */
export async function getTaskStatus(taskId: string): Promise<StoryboardTask> {
  const response = await api.get(`/storyboard/tasks/${taskId}`)
  return response.data
}

/**
 * 获取章节的最新任务
 */
export async function getChapterTask(chapterId: string): Promise<StoryboardTask | null> {
  const response = await api.get(`/storyboard/chapters/${chapterId}/task`)
  return response.data
}

/**
 * 轮询分镜任务直到完成
 * @param taskId 任务ID
 * @param onProgress 进度回调
 * @param interval 轮询间隔（毫秒）
 * @param maxAttempts 最大尝试次数
 */
export async function pollStoryboardTaskUntilComplete(
  taskId: string,
  onProgress?: (task: StoryboardTask) => void,
  interval: number = 1500,
  maxAttempts: number = 120
): Promise<StoryboardTask> {
  let attempts = 0

  while (attempts < maxAttempts) {
    const task = await getTaskStatus(taskId)

    if (onProgress) {
      onProgress(task)
    }

    if (task.status === 'completed' || task.status === 'failed' || task.status === 'cancelled') {
      return task
    }

    attempts++
    await new Promise(resolve => setTimeout(resolve, interval))
  }

  throw new Error('Task polling timeout')
}

// ============== 分镜数据 API 函数 ==============

/**
 * 获取章节的分镜脚本
 */
export async function getStoryboard(chapterId: string): Promise<StoryboardResponse> {
  const response = await api.get(`/storyboard/chapters/${chapterId}`)
  return response.data
}

/**
 * 更新单个分镜
 */
export async function updateShot(
  chapterId: string,
  sequence: number,
  update: ShotUpdateRequest
): Promise<Shot> {
  const response = await api.put(`/storyboard/chapters/${chapterId}/shots/${sequence}`, update)
  return response.data
}

/**
 * 删除单个分镜
 */
export async function deleteShot(
  chapterId: string,
  sequence: number
): Promise<{ message: string; shot_count: number }> {
  const response = await api.delete(`/storyboard/chapters/${chapterId}/shots/${sequence}`)
  return response.data
}

/**
 * 添加新分镜
 */
export async function addShot(
  chapterId: string,
  shot: Partial<Shot>,
  afterSequence?: number
): Promise<Shot> {
  const params = afterSequence !== undefined ? { after_sequence: afterSequence } : {}
  const response = await api.post(`/storyboard/chapters/${chapterId}/shots`, shot, { params })
  return response.data
}

/**
 * 获取分镜的视频生成提示词
 */
export async function getStoryboardPrompts(
  chapterId: string,
  platform: 'vidu' | 'doubao' | 'veo' | 'kling' | 'sora' = 'vidu'
): Promise<StoryboardPromptsResponse> {
  const response = await api.get(`/storyboard/chapters/${chapterId}/prompts`, {
    params: { platform },
  })
  return response.data
}

// ============== 枚举值常量（用于前端选择器）==============

export const SHOT_SIZES = [
  { value: 'extreme_close_up', label: '特写', labelEn: 'Extreme Close-up' },
  { value: 'close_up', label: '近景', labelEn: 'Close-up' },
  { value: 'medium_close_up', label: '中近景', labelEn: 'Medium Close-up' },
  { value: 'medium_shot', label: '中景', labelEn: 'Medium Shot' },
  { value: 'cowboy_shot', label: '牛仔镜头', labelEn: 'Cowboy Shot' },
  { value: 'medium_long_shot', label: '中远景', labelEn: 'Medium Long Shot' },
  { value: 'full_shot', label: '全景', labelEn: 'Full Shot' },
  { value: 'long_shot', label: '远景', labelEn: 'Long Shot' },
  { value: 'extreme_long_shot', label: '大远景', labelEn: 'Extreme Long Shot' },
  { value: 'establishing_shot', label: '建立镜头', labelEn: 'Establishing Shot' },
]

export const CAMERA_ANGLES = [
  { value: 'eye_level', label: '平视', labelEn: 'Eye Level' },
  { value: 'low_angle', label: '仰拍', labelEn: 'Low Angle' },
  { value: 'high_angle', label: '俯拍', labelEn: 'High Angle' },
  { value: 'birds_eye', label: '鸟瞰', labelEn: "Bird's Eye" },
  { value: 'worms_eye', label: '蚂蚁视角', labelEn: "Worm's Eye" },
  { value: 'dutch_angle', label: '荷兰角', labelEn: 'Dutch Angle' },
  { value: 'over_the_shoulder', label: '过肩', labelEn: 'Over the Shoulder' },
  { value: 'pov', label: '第一人称', labelEn: 'POV' },
]

export const CAMERA_MOVEMENTS = [
  { value: 'static', label: '静态', labelEn: 'Static' },
  { value: 'pan_left', label: '左摇', labelEn: 'Pan Left' },
  { value: 'pan_right', label: '右摇', labelEn: 'Pan Right' },
  { value: 'tilt_up', label: '上摇', labelEn: 'Tilt Up' },
  { value: 'tilt_down', label: '下摇', labelEn: 'Tilt Down' },
  { value: 'dolly_in', label: '推镜头', labelEn: 'Dolly In' },
  { value: 'dolly_out', label: '拉镜头', labelEn: 'Dolly Out' },
  { value: 'truck_left', label: '左移', labelEn: 'Truck Left' },
  { value: 'truck_right', label: '右移', labelEn: 'Truck Right' },
  { value: 'tracking', label: '跟踪', labelEn: 'Tracking' },
  { value: 'orbit', label: '环绕', labelEn: 'Orbit' },
  { value: 'handheld', label: '手持', labelEn: 'Handheld' },
  { value: 'steadicam', label: '稳定器', labelEn: 'Steadicam' },
  { value: 'crane_up', label: '升镜头', labelEn: 'Crane Up' },
  { value: 'crane_down', label: '降镜头', labelEn: 'Crane Down' },
  { value: 'whip_pan', label: '急摇', labelEn: 'Whip Pan' },
  { value: 'push_in', label: '缓推', labelEn: 'Push In' },
  { value: 'pull_back', label: '缓拉', labelEn: 'Pull Back' },
]

export const VIDEO_STYLES = [
  { value: 'cinematic', label: '电影感', labelEn: 'Cinematic' },
  { value: 'photorealistic', label: '照片写实', labelEn: 'Photorealistic' },
  { value: 'anime', label: '动漫', labelEn: 'Anime' },
  { value: 'cartoon', label: '卡通', labelEn: 'Cartoon' },
  { value: 'noir', label: '黑色电影', labelEn: 'Noir' },
  { value: 'vintage', label: '复古', labelEn: 'Vintage' },
  { value: 'documentary', label: '纪录片', labelEn: 'Documentary' },
  { value: 'fantasy', label: '奇幻', labelEn: 'Fantasy' },
  { value: 'sci_fi', label: '科幻', labelEn: 'Sci-Fi' },
]

export const MOODS = [
  { value: 'peaceful', label: '平和', labelEn: 'Peaceful' },
  { value: 'tense', label: '紧张', labelEn: 'Tense' },
  { value: 'romantic', label: '浪漫', labelEn: 'Romantic' },
  { value: 'melancholic', label: '忧郁', labelEn: 'Melancholic' },
  { value: 'mysterious', label: '神秘', labelEn: 'Mysterious' },
  { value: 'joyful', label: '欢快', labelEn: 'Joyful' },
  { value: 'dramatic', label: '戏剧性', labelEn: 'Dramatic' },
  { value: 'epic', label: '史诗', labelEn: 'Epic' },
  { value: 'intimate', label: '亲密', labelEn: 'Intimate' },
  { value: 'suspenseful', label: '悬疑', labelEn: 'Suspenseful' },
]

export const LIGHTING_STYLES = [
  { value: 'natural', label: '自然光', labelEn: 'Natural' },
  { value: 'golden_hour', label: '黄金时刻', labelEn: 'Golden Hour' },
  { value: 'blue_hour', label: '蓝调时刻', labelEn: 'Blue Hour' },
  { value: 'high_key', label: '高调光', labelEn: 'High Key' },
  { value: 'low_key', label: '低调光', labelEn: 'Low Key' },
  { value: 'dramatic', label: '戏剧光', labelEn: 'Dramatic' },
  { value: 'backlit', label: '逆光', labelEn: 'Backlit' },
  { value: 'neon', label: '霓虹', labelEn: 'Neon' },
  { value: 'moonlight', label: '月光', labelEn: 'Moonlight' },
  { value: 'volumetric', label: '体积光', labelEn: 'Volumetric' },
]

export const ASPECT_RATIOS = [
  { value: '16:9', label: '16:9 横屏', labelEn: '16:9 Landscape' },
  { value: '9:16', label: '9:16 竖屏', labelEn: '9:16 Portrait' },
  { value: '1:1', label: '1:1 方形', labelEn: '1:1 Square' },
  { value: '21:9', label: '21:9 超宽', labelEn: '21:9 Ultrawide' },
  { value: '2.39:1', label: '2.39:1 宽银幕', labelEn: '2.39:1 Anamorphic' },
]

export const TARGET_PLATFORMS = [
  { value: 'vidu', label: 'Vidu Q2', description: '参考图格式 @id，细腻表情、平滑镜头运动' },
  { value: 'doubao', label: '豆包 Seedance', description: '参考图格式 [图N]，火山引擎视频模型' },
]
